packages/amplify-cli/src/commands/push.ts

```diff
@@ -1,21 +1,13 @@
-import {
-  $TSAny,
-  $TSContext,
-  AmplifyError,
-  AmplifyFault,
-  spinner,
-  stateManager,
-} from 'amplify-cli-core';
+import { $TSAny, $TSContext, AmplifyError, AmplifyFault, spinner, stateManager } from 'amplify-cli-core';
 import sequential from 'promise-sequential';
 import {
   notifyFieldAuthSecurityChange,
   notifyListQuerySecurityChange,
   notifySecurityEnhancement,
 } from '../extensions/amplify-helpers/auth-notifications';
-import {
-  getProviderPlugins,
-} from '../extensions/amplify-helpers/get-provider-plugins';
+import { getProviderPlugins } from '../extensions/amplify-helpers/get-provider-plugins';
 import { updateCognitoTrackedFiles } from '../extensions/amplify-helpers/update-tracked-files';
+import runBuildCommand from '../utils/run-build-command';
```

Comment:
I'm not sure this is needed, pull is executed before every push. Can you verify that runBuildCommand is only being run once during `amplify push`?

Reply:
I can confirm running `amplify-dev push --runBuildCommand` resulted in the build command running only once.

It looks like `amplify push` is not using the `pullBackend` command. So it is pulling, but through a different code path than where `runBuildCommand` is located.

I could move the `runBuildCommand` to that path if you think that would be better.

Reply:
No I think this is fine, I just wanted to be sure we weren't going to rebuild unnecessarily. Thanks!